Heimdallr/Core/Heimdallr.swift

```diff
@@ -170,6 +167,24 @@ public let HeimdallrErrorNotAuthorized = 2
             }
         }
     }
+    
+    /// Updates the stored access token with a new one.
+    ///
+    /// - parameter accessToken: The new access token.
+    ///
+    /// - returns: The updated access token.
+    private func updateAccessToken(accessToken: OAuthAccessToken) -> OAuthAccessToken {
```

Comment:
I think I would prefer this being part of the `set` part of `accessToken` just so that we always get the expected behaviour when setting a new token directly. What do you think?